src/main/java/uk/gov/pay/api/exception/mapper/CreateRefundExceptionMapper.java

```diff
@@ -33,15 +35,35 @@ public Response toResponse(CreateRefundException exception) {
             requestError = aRequestError(CREATE_PAYMENT_REFUND_NOT_FOUND_ERROR);
             status = NOT_FOUND;
 
-        } else if (exception.getErrorIdentifier() == ErrorIdentifier.REFUND_NOT_AVAILABLE && exception.hasReason()) {
-            requestError = aRequestError(CREATE_PAYMENT_REFUND_NOT_AVAILABLE, exception.getReason());
-            status = BAD_REQUEST;
-        } else if (exception.getErrorIdentifier() == ErrorIdentifier.REFUND_AMOUNT_AVAILABLE_MISMATCH) {
-            requestError = aRequestError(CREATE_PAYMENT_REFUND_AMOUNT_AVAILABLE_MISMATCH);
-            status = PRECONDITION_FAILED;
-        } else {
-            requestError = aRequestError(CREATE_PAYMENT_REFUND_CONNECTOR_ERROR);
-            status = INTERNAL_SERVER_ERROR;
+        }
+        else {
+            switch (exception.getErrorIdentifier()) {
+                case ACCOUNT_DISABLED: {
+                    requestError = aRequestError(ACCOUNT_DISABLED);
+                    status = FORBIDDEN;
+                    break;
+                }
+                case REFUND_NOT_AVAILABLE: {
+                    if (exception.hasReason()) {
+                        requestError = aRequestError(CREATE_PAYMENT_REFUND_NOT_AVAILABLE, exception.getReason());
+                        status = BAD_REQUEST;
+                    }
+                    else {
```

Comment:
Maybe we should log something explicit here at error level saying the reason was missing, as we never expect this to happen